### PR TITLE
Android build on travis

### PR DESCRIPTION
**Pull request by samgiles**
Trying travis.

### CODE REVIEW

**Comment by wilsonpage** [comment]
Loookin' gouuuud! Any reason why we're not using `react-native` to build?

**Comment by samgiles** [comment]
Because I haven't got that far yet, but I _think_ react native has put everything necessary in the gradle build config anyway.

**Comment by wilsonpage** [comment]
Sick as fuck :sunglasses:

**Comment by samgiles** [comment]
r? @wilsonpage 
Just builds an android release APK, doesn't do anything with it yet, just asserts that we can and nothing is broken.

We need an Apple developer account so we can code sign the iOS release, so I've disabled that for now.

File: build-scripts/bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -e
+
+echo "Installing dependencies from brew.."
+# Install tools
+brew reinstall gradle flow watchman xctool
```
Comment (wilsonpage) [inline build-scripts/bootstrap.sh]: I don't think `watchman` is required for production builds. Just monitoring JS files for changes and re-bundling when in development.

File: build-scripts/bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+echo "Installing react-native.."
+# Install react-native
+npm install -g react-native-cli
```
Comment (wilsonpage) [inline build-scripts/bootstrap.sh:8]: Where is this used?

File: build-scripts/build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+
+if [[ "$BUILD_TYPE" == "android" ]];then
+    cd android
+    ./gradlew assembleRelease | tee
```
Comment (wilsonpage) [inline build-scripts/build.sh:7]: Does this spit out the `.apk` somewhere?

File: build-scripts/bootstrap.sh (hunk 2 above)
Comment (samgiles) [inline build-scripts/bootstrap.sh:8]: The tools are used internally by the build for each platform. Running `./gradlew assembleRelease` or the equivalent `xcodebuild` command, fails without these tools available.

File: build-scripts/build.sh (hunk 3 above)
Comment (samgiles) [inline build-scripts/build.sh:7]: Yup, you end up with "magnet/android/app/build/outputs/apk/*.apk" after this PR I'll get them pushed to Github (or some other storage).

**Comment by wilsonpage** [comment]
@samgiles let's ship it then. Feel free to land :)

**Pull request merged**